src/Magento/FunctionalTestingFramework/Suite/SuiteGenerator.php

```diff
@@ -140,35 +149,76 @@ public function generateSuite($suiteName)
      * @param array  $tests
      * @param string $originalSuiteName
      * @return void
-     * @throws TestReferenceException
-     * @throws XmlException
-     * @throws TestFrameworkException
+     * @throws \Exception
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      */
     private function generateSuiteFromTest($suiteName, $tests = [], $originalSuiteName = null)
     {
         $relativePath = TestGenerator::GENERATED_DIR . DIRECTORY_SEPARATOR . $suiteName;
         $fullPath = FilePathFormatter::format(TESTS_MODULE_PATH) . $relativePath . DIRECTORY_SEPARATOR;
 
         DirSetupUtil::createGroupDir($fullPath);
+        $exceptionCollector = new ExceptionCollector();
+        try {
+            $relevantTests = [];
+            if (!empty($tests)) {
+                $this->validateTestsReferencedInSuite($suiteName, $tests, $originalSuiteName);
+                foreach ($tests as $testName) {
+                    try {
+                        $relevantTests[$testName] = TestObjectHandler::getInstance()->getObject($testName);
+                    } catch (FastFailException $e) {
+                        throw $e;
+                    } catch (\Exception $e) {
+                        $exceptionCollector->addError(
+                            self::class,
+                            "Unable to find relevant test \"{$testName}\" for suite \"{$suiteName}\""
+                        );
+                    }
+                }
+            } else {
+                $relevantTests = SuiteObjectHandler::getInstance()->getObject($suiteName)->getTests();
+            }
 
-        $relevantTests = [];
-        if (!empty($tests)) {
-            $this->validateTestsReferencedInSuite($suiteName, $tests, $originalSuiteName);
-            foreach ($tests as $testName) {
-                $relevantTests[$testName] = TestObjectHandler::getInstance()->getObject($testName);
+            if (empty($relevantTests)) {
+                $exceptionCollector->reset();
+                throw new TestFrameworkException("No relevant test found for suite \"{$suiteName}\"");
             }
-        } else {
-            $relevantTests = SuiteObjectHandler::getInstance()->getObject($suiteName)->getTests();
-        }
 
-        $this->generateRelevantGroupTests($suiteName, $relevantTests);
-        $groupNamespace = $this->generateGroupFile($suiteName, $relevantTests, $originalSuiteName);
+            try {
+                $this->generateRelevantGroupTests($suiteName, $relevantTests);
+            } catch (FastFailException $e) {
+                throw $e;
+            } catch (\Exception $e) {
+                $exceptionCollector->addError(
+                    self::class,
+                    "Failed to generate tests for suite \"{$suiteName}\""
```

Comment:
I introduced a bad entity reference in a test that is a part of WYSIWYGDisabledSuite, but I don't see this error message on the console. mftf.log looks good to me, and the test did not generate, which looks good to.

```
[2020-09-24 15:47:23] Magento\FunctionalTestingFramework\Util\TestGenerator.ERROR: Failed to generate NewProductsListWidgetBundleProductTest [] []
[2020-09-24 15:47:23] Magento\FunctionalTestingFramework\Test\Handlers\TestObjectHandler.ERROR: Removed invalid test object NewProductsListWidgetBundleProductTest [] []
```

Reply:
what is the expected behavior if we try to generate a non-existant suite
I ran this:
`bin/mftf generate:suite NonExistant suite`
What I see is all tests are generated except for the ones with errors. That shouldn't be the case, should it? On `develop` the command throws an error, and no tests get generated.

Reply:
I cannot reproduce this one it seems. I got 
> ERROR: 2 Test(s) generated with annotation errors. See mftf.log for details.
> ERROR: 1 Suite(s) failed to generate. See mftf.log for details.
> No Suite Generated

Can you specify the steps?

Reply:
> I introduced a bad entity reference in a test that is a part of WYSIWYGDisabledSuite, but I don't see this error message on the console. mftf.log looks good to me, and the test did not generate, which looks good to.
> 
> ```
> [2020-09-24 15:47:23] Magento\FunctionalTestingFramework\Util\TestGenerator.ERROR: Failed to generate NewProductsListWidgetBundleProductTest [] []
> [2020-09-24 15:47:23] Magento\FunctionalTestingFramework\Test\Handlers\TestObjectHandler.ERROR: Removed invalid test object NewProductsListWidgetBundleProductTest [] []
> ```

This is expected and result of the change. When a bad test is encountered, it's removed from the test objects and it's not discovered by the suite.

Reply:
yes, but shouldn't it increment the number of tests(s) failed to generate in the console message?

Reply:
> I cannot reproduce this one it seems. I got
> 
> > ERROR: 2 Test(s) generated with annotation errors. See mftf.log for details.
> > ERROR: 1 Suite(s) failed to generate. See mftf.log for details.
> > No Suite Generated
> 
> Can you specify the steps? And there is no test or suite generated.

I am unable to as well. Maybe some tests that were left behind in a previous run

Reply:
> yes, but shouldn't it increment the number of tests(s) failed to generate in the console message?
Only part left to verify is the above, from these comments

Reply:
The bad test will be removed and counted for a failure, but the suite will not know there is this failing test. This is ok.